### ISSUE TITLE
can't reproduce examples

### ISSUE THREAD

**Issue by casanovarodrigo** [broidHQ/broid-kit#21]
Hello. I didn't know where to ask this, sorry if this isn't the place for it.

I'm trying to reproduce the examples on the readme file, but despite my efforts, I cant. I first created an app on heroku, then created a simple bot directly using methods that fb gives in its documentation. Now i'm trying to make broid-kit work but looks like i'm missing something.

index.js on heroku

```
const Bot = require("@broid/kit");
const BroidMessenger = require("@broid/messenger");

const token = process.env.FB_PAGE_ACCESS_TOKEN
const vtoken = process.env.FB_VERIFY_ACCESS_TOKEN

const express = require('express')
const app = express()

const bot = new Bot({
  logLevel: "info"
});

bot.use(new BroidMessenger({token: token, tokenSecret: vtoken}));

app.use("/", bot.getRouter());

app.listen(process.env.PORT || 5000);

// Listening for public starting by regex match for `hello`
bot.hear(".*", "Group")
  .subscribe((data) => {
    console.log("Data:", JSON.stringify(data, null, 2));

    // Reply to the message
    bot.sendText("Hi, How are you?", data.message);
  });
```

This should return a message at every message sent, right?

**Comment by dustinblackman** [comment]
Yep it looks like that should work just fine, if you're in group messages. If you'd like to listen for both Group and Personal messages for example, you can do something like this.

```javascript
const Observable = require('rxjs').Observable;

Observable.merge(bot.on('Group'), bot.on('Person'))
  .subscribe((data) => {....})
```

We'll be adding a method in the future as something like `all` or similar. Also, were you able to setup your Facebook application correctly where it was able to authorize with Broid?

**Comment by casanovarodrigo** [comment]
Huh. Tried adding this lines, no difference though. Yeah, the fb app authtorize when it hits the webhook, it's `webhook/messenger` right? And, when I send a chat message, messenger sends the event to the webhook, but is not sending back any message. Heroku's log when I send a massage to the bot.

`
2017-07-04T05:32:57.126294+00:00 heroku[router]: at=info method=POST path="/webhook/messenger" host=test-tbot.herokuapp.com request_id=85c17d7c-f48f-49f3-9e61-5f7e3fbf4ae9 fwd="173.252.88.191" dyno=web.1 connect=1ms service=84ms status=200 bytes=201 protocol=https`

Also, when I deploy, heroku logs this. For what I've seen of broidkit's code, looks like just a building log
`2017-07-04T05:31:20.574964+00:00 app[web.1]: {"pid":4,"hostname":"f906d43a-eaf8-4f57-943f-c95ac6ab8862","name":"broidkit","level":30,"time":1499146280571,"method":"use","message":"Integration: messenger","v":1}`

I made the repository public if you want to check it out. I think I have all the dependencies.

[https://github.com/casanovarodrigo/trialbot](https://github.com/casanovarodrigo/trialbot)

**Comment by killix** [comment]
This code should work. Do you have any other logs/info to help you?

**Comment by casanovarodrigo** [comment]
Sadly I don't have anymore logs

Started to think that could be something with heroku. Could it be? huh
I'll try other host system by tomorrow

**Comment by killix** [comment]
thanks, we will check. I think it's something around this log line:

```
2017-07-05T08:59:29.506280+00:00 heroku[router]: at=error code=H12 desc="Request timeout" method=GET path="/webhook/messenger" host=test-tbot.herokuapp.com request_id=bb06040a-6e6a-459e-8ed5-23abf5275c77 fwd="143.54.171.119" dyno=web.1 connect=0ms service=30001ms status=503 bytes=0 protocol=http
```

I understand better the issue, it's something around this line: [L221](https://github.com/broidHQ/integrations/blob/master/broid-messenger/src/core/Adapter.ts#L221)

Can you please try to log the `req` body directly on the `@broid/messenger` lib? 
Or if you can't explain us the exact step to try to repro in our side the issue.

**Comment by casanovarodrigo** [comment]
Hey!

I gave up on heroku and started to https tunneling my localhost env, much better. Now I have a lot of logs for you :)

Here's the log on **req** at the facebook auth webhook, when I first create the webhook on fb dev page.
[Webhok log](https://pastebin.com/4sz0PVY2)

I also added a console.log(req) at the `router.post` [L232](https://github.com/broidHQ/integrations/blob/master/broid-messenger/src/core/Adapter.ts#L232) method, so it would give me infos of the `req` when I received a message, it doesn't log anything though.

**Comment by casanovarodrigo** [comment]
Today I tried to replicate it in another enviroment, this time at my workplace.

[Here is the log](https://pastebin.com/4CAySQ4E) of the req prop in router.post method. 

I tought it had something to do with the fb app privileges so I checked every permission box possible and still the same problem.

**Comment by fdematos** [comment]
Hi, I got the same issue as you when deploying on Heroku.
To make it run on heroku just remove the host parameters : 

const bot = new Bot({
  logLevel: "trace",
  http: {
    port: process.env.PORT
  }
});
